truss/templates/triton/model/transform.py

```diff
@@ -0,0 +1,78 @@
+import json
+
+import numpy as np
+from utils.pydantic import PYTHON_TYPE_TO_NP_DTYPE, inspect_pydantic_model
+from utils.triton import (
+    convert_tensor_to_python_type,
+    create_inference_response,
+    create_tensor,
+    get_input_tensor_by_name,
+)
+
+
+def transform_triton_to_pydantic(triton_requests, pydantic_type) -> list:
+    """
+    Transforms a list of Triton requests into a list of Pydantic objects.
+
+    This function inspects the fields of the given Pydantic type and extracts
+    the corresponding tensors from the Triton requests. It then converts these
+    tensors into Python types and uses them to instantiate the Pydantic objects.
+
+    Args:
+        triton_requests (list): A list of Triton request objects.
+        pydantic_type (Type[BaseModel]): The Pydantic model type to which the Triton requests should be transformed.
+
+    Returns:
+        list: A list of Pydantic objects instantiated from the Triton requests.
+
+    Raises:
+        ValueError: If a tensor corresponding to a Pydantic field cannot be found in the Triton request.
+    """
+    fields = inspect_pydantic_model(pydantic_type)
+    results = []
+
+    for request in triton_requests:
+        data = {}
+        for field in fields:
+            field_name, field_type = field.values()
+            tensor = get_input_tensor_by_name(request, field_name)
+            data[field_name] = convert_tensor_to_python_type(tensor, field_type)
+        results.append(pydantic_type(**data))
+
+    return results
+
+
+def transform_pydantic_to_triton(pydantic_objects):
+    """
+    Transforms a list of Pydantic objects into a list of Triton inference responses.
+
+    This function iterates over the fields of each Pydantic object, determines the
+    appropriate tensor data type, and creates a tensor for each field. These tensors
+    are then used to create Triton inference responses.
+
+    Args:
+        pydantic_objects (list): A list of Pydantic objects to be transformed.
+
+    Returns:
+        list: A list of Triton inference responses created from the Pydantic objects.
+
+    Raises:
+        ValueError: If a Python type in the Pydantic object does not have a corresponding numpy dtype.
+    """
+    results = []
+
+    for obj in pydantic_objects:
```

Comment:
So -- the user could theoretically return anything from the `predict` function in practice. There's nothing _really_ preventing them from returning garbage.

If they do that, this might return an unhelpful exception right? Should we do some validation on the outputs of the user's predict function?

Reply:
There's a couple layers to this. There's the Pydantic validation that occurs (so a user can't try to set a field with a value that is not of the appropriate type). We also (now) validate that the user is returning a list of Pydantic objects and if not, try to error out with a helpful error. Good call out.